### PR TITLE
Rebased mainline

### PR DESCRIPTION
**Pull request by mhhf**
new pr making everything else look old

### CODE REVIEW

File: lib/compile_BACKUP_24976.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+  const rules = logs
+    .filter(l => l.split(" ")[1] == "RULE")
+    .map(l => l.split(" ")[2])
+    .reduce((a, ruleid) => {
+<<<<<<< HEAD
+      if(!a[ruleid]) a[ruleid] = JSON.parse(read(path.join(KLAB_WD_PATH, `${proofid}_blobs`, ruleid + ".json")));
+=======
+      if(!a[ruleid]) a[ruleid] = JSON.parse(read(path.join(KLAB_WD_PATH, proofid + "_blobs", ruleid + ".json")));
+>>>>>>> major mainline k rewrk, wip but working version
```
Comment (ehildenb) [inline lib/compile_BACKUP_24976.js]: Should these be here?

File: resources/run_BACKUP_29483.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+=======
+K_OPTS=-Xmx10G $KLAB_EVMS_PATH/.build/k/k-distribution/target/release/k/bin/kprove \
+    --debugg --debugg-path $TMPDIR/klab --debugg-id $2 \
+    --directory $KLAB_EVMS_PATH/.build/java/ --def-module RULES \
+    --z3-executable --smt_prelude ./prelude.smt2 --z3-tactic "(or-else (using-params smt :random-seed 3 :timeout 1000) (using-params smt :random-seed 2 :timeout 2000) (using-params smt :random-seed 1))" $1 \
+    --output json \
+    --output-tokenize "#And _==K_ <k> #unsigned" \
+    --output-omit "<programBytes> <program> <code> <previousGas> <touchedAccounts> <interimStates> <callStack>" \
+    --output-flatten "_Map_ #And"
+>>>>>>> resources/run.sh: newlines for readability
```
Comment (ehildenb) [inline resources/run_BACKUP_29483.sh]: Should these be here?

File: lib/pull.js [hunk 3]
```diff
@@ -98,90 +104,90 @@ const pull = ({K, onion}) => {
...
+  //     // return {
+  //     //   ...state,
+  //     //   behaviour: {
+  //     //     ...state.behaviour,
+  //     //     frontier: frontier_,
+  //     //     height: height_,
+  //     //     root: propagate(state.behaviour.root)
+  //     //   },
+  //     // };
+  //   })
```
Comment (MrChico) [inline lib/pull.js]: Can this be removed completely?

File: .gitignore [hunk 4]
```diff
@@ -5,3 +5,4 @@
 /examples/*/out/
 /haskell/resultlibexec/klab-gas-analyser
 haskell/result
+/examples/*/out/*
```
Comment (MrChico) [inline .gitignore]: reduntant, line 5 takes care of this

**Comment by MrChico** [comment]
reactive functionality does not seem to work in this edition. While my proof is running I cannot see any steps at all (even after restarting klab client)

**Comment by MrChico** [comment]
Also, doesn't seem to be able to debug `Safeadd_add_fail.k` nor a proof that is rejected (such as `Token_transfer_pass_rough.k`). Seems like something is wrong in the `genBehaviour` functionality

**Comment by livnev** [comment]
@MrChico you need to push your new version of `evm-semantics` to `dapphub/evm-semantics` otherwise no one else can find it.

**Pull request closed without merging**